library/core/src/option.rs

```diff
@@ -689,6 +689,92 @@ impl<T> Option<T> {
         }
     }
 
+    const fn get_some_offset() -> isize {
+        if core::mem::size_of::<Option<T>>() == core::mem::size_of::<T>() {
```

Comment:
nit: since we're *in* core, we usually don't call it `core`, IIRC.  `crate::mem` is already imported here (line 510), so I think this should be written

```suggestion
        if mem::size_of::<Option<T>>() == mem::size_of::<T>() {
```

(and ditto other places below)